### PR TITLE
filter daml internal modules and declarations

### PR DESCRIPTION
**Pull request by ghost**
This adds an additional filter on daml internal modules and declarations.

### CODE REVIEW

**Comment by ghost** [comment]
Not a 100% sure where the filter for `_choice_` needs to go.

File: src/Development/IDE/LSP/Outline.hs [hunk 1]
```diff
@@ -205,6 +208,9 @@ documentSymbolForImport (L l ImportDecl { ideclName, ideclQualified, ideclImplic
 documentSymbolForImport (L _ XImportDecl {}) = Nothing
 #endif
 
+isDamlInternalImport :: ModuleName -> Bool
+isDamlInternalImport ideclName = "DA.Internal" `T.isPrefixOf` pprText ideclName
```
Comment (cocreature) [inline src/Development/IDE/LSP/Outline.hs]: should we limit this to `DA.Internal.Desugar`? I’m a bit worried we have some useful stuff in `DA.Internal` that we reexport from other non-internal modules.

**Review by cocreature — APPROVED** [review]
Thanks! We should test for this on the daml side. I would suggest to keep this open until we have a passing test there.

File: src/Development/IDE/LSP/Outline.hs (hunk 1 above)
Comment (ghost) [reply src/Development/IDE/LSP/Outline.hs]: Maybe at least `DA.Internal.Desugar` and `DA.Internal.Record`.

Comment (cocreature) [reply src/Development/IDE/LSP/Outline.hs]: sounds good :+1:

**Comment by ghost** [comment]
Tested this manually in daml studio, looks good.

**Pull request merged**